Classes/common/CDTDatastore.m

```diff
@@ -63,6 +64,7 @@ - (id)initWithDatabase:(TD_Database *)database
         NSString *name = [database name];
         _extensionsDir =
             [dir stringByAppendingPathComponent:[NSString stringWithFormat:@"%@_extensions", name]];
+        _manager = manager;
```

Comment:
OK, I see your point. I can not even find a unit test that creates a `CDTDatastore` but in the [branch](https://github.com/cloudant/CDTDatastore/tree/feature-encryption) I'm working on right now, I also modified this `init` method to supply a key provider and I ensure that the provider is always informed with an `Assert`. I also added a few tests that create a `CDTDatastore` without using a `CDTDatastoreManager` and check that I receive an exception if the provider is not informed.

I think we need a second opinion here, @mikerhodes what do you think?